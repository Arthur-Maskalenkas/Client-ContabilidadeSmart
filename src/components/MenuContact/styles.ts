import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    border-bottom: 1px solid ${theme.colors.darkGray};
  `}
`

export const ContactInfoWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    padding-right: 2rem;
    padding-left: 2rem;

    height: 100%;

    &:nth-child(2) {
      border-right: 1px solid ${theme.colors.darkGray};
      border-left: 1px solid ${theme.colors.darkGray};
    }

    ${IconWrapper}, ${ContactInfo} {
      cursor: pointer;
      transition: 0.5s all;
    }

    ${IconWrapper}:hover {
      background-color: ${theme.colors.secondary};
    }
    ${ContactInfo}:hover {
      color: ${theme.colors.secondary};
    }
  `}
`

export const ContactInfo = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};

    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.family.poppins.weight.light};
  `}
`

export const IconWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    padding: 1.2rem;
    border-radius: 50%;

    margin-right: ${theme.spacings.xxsmall};
    background: rgb(255 255 255 / 9%);

    & > svg {
      width: 2rem;
      color: white;
    }
  `}
`

export const WrapperContent = styled.div`
  position: relative;
  height: 7.5rem;
  display: flex;
`
