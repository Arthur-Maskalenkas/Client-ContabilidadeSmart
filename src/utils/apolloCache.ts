import { InMemoryCache } from '@apollo/client'
import { concatPagination } from '@apollo/client/utilities'

// A query QueryGames retorna -> games <-, e é isso que o cache vai acumular games

// ** fluxo:
// > Eu pego uma lista de free (retorna games assim como varias outras query)
// > o cache fica [game1,game2,game3]
// > Assim que outra parte usa alguma query que retorne game, ela vai receber independente de tudo [game1,game2,game3], e não [game4,game5,game6]
// **

// Para quebrar o fluxo, eu separei o cacheamento por 'where' e 'sort'. Assim fica algo como: (caso puxe 3 games de 3 queries diferentes)

//**
// > games: {"where: {"release_date_lte":"2021-01-29"}, "sort":"release_date:desc"}: [] 8 items"}
// > games: {"where: {"release_date_gt":"2021-01-29"}, "sort":"release_date:asc"}: [] 8 items"}
// > games: {"where: {"price": 0}}: [] 8 items
// > Antes ficaria tudo em games com 24 items
//**
export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        posts: concatPagination(['where', 'sort'])
      }
    }
  }
})
