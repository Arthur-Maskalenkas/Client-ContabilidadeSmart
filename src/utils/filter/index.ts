import { ParsedUrlQueryInput } from 'querystring'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
}

export const parseQueryStringToWhere = ({ queryString }: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  // queryString = {platforms: ['windows','linux']}
  // filterItems = {name: 'platforms', type: 'checkbox'}
  Object.keys(queryString).forEach((key) => {
    // Pegando o valor e colocando dentro do objeto
    obj[key] = { slug_contains: queryString[key] }
  })

  return obj
}
