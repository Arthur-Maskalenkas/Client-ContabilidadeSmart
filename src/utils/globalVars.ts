export const IMAGE_NOT_FOUND =
  'https://res.cloudinary.com/dezwlfeyb/image/upload/v1635853776/102-022_800x_ebxk1t.jpg'

export const LINK_SITE = 'localhost:3000'

// --  API

export const API_LINKEDIN =
  'https://www.linkedin.com/shareArticle?mini=true&url='

export const API_FACEBOOK = 'https://www.facebook.com/sharer/sharer.php?u='

export const API_WHATSAPP = 'https://api.whatsapp.com/send?text='

// --  API
