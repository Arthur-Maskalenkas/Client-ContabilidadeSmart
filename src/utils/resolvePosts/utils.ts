export const operations = {
  nextPage: nextPageOperation(),
  backPage: backPageOperation()
}

type variablesType = {
  limit?: number
  start?: number
}

export type FetchMoreProps = {
  variables: variablesType
}
