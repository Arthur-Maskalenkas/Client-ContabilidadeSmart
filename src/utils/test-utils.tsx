import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const customRender = (ui: ReactElement, { ...renderOptions }: CustomRenderProps = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

// Tudo que eu chamar, vai vim dessa pasta aqui
export * from '@testing-library/react'
export { customRender as render }
