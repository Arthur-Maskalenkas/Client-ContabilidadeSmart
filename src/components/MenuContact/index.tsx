import * as S from './styles'

import { Instagram, Whatsapp } from '@styled-icons/bootstrap/'
import { Gmail } from '@styled-icons/simple-icons/Gmail'
import { Container } from 'components/Container'

const MenuContact = () => (
  <S.Wrapper>
    <Container>
      <S.WrapperContent>
        <S.ContactInfoWrapper>
          <S.IconWrapper>
            <Whatsapp title="whatsapp" role="svg" />
          </S.IconWrapper>
          <S.ContactInfo>+55 17 99146-0296</S.ContactInfo>
        </S.ContactInfoWrapper>

        <S.ContactInfoWrapper>
          <S.IconWrapper>
            <Gmail title="Gmail" role="svg" />
          </S.IconWrapper>
          <S.ContactInfo>contato@contabilidadesmart.com.br</S.ContactInfo>
        </S.ContactInfoWrapper>

        <S.ContactInfoWrapper>
          <S.IconWrapper>
            <Instagram title="Instagram" role="svg" />
          </S.IconWrapper>
          <S.ContactInfo>@contabilidadesmart</S.ContactInfo>
        </S.ContactInfoWrapper>
      </S.WrapperContent>
    </Container>
  </S.Wrapper>
)

export default MenuContact
