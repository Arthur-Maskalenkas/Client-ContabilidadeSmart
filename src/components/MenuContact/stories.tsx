import { Story, Meta } from '@storybook/react'
import MenuContact from '.'

import { MINIMAL_VIEWPORTS } from '@storybook/addon-viewport'

export default {
  title: 'menu/MenuContact',
  component: MenuContact
} as Meta

export const Default: Story = () => <MenuContact />

Default.args = {}

Default.parameters = {
  backgrounds: {
    default: 'mm-dark'
  }
}

export const Mobile: Story = () => <MenuContact />

Mobile.args = {}

Mobile.parameters = {
  backgrounds: {
    default: 'mm-light'
  },
  viewport: {
    viewports: MINIMAL_VIEWPORTS,
    defaultViewport: 'mobile1'
  }
}
