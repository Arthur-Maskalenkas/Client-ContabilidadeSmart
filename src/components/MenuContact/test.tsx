import { render, screen } from 'utils/test-utils'

import MenuContact from '.'

describe('<MenuContact />', () => {
  it('should render the heading', () => {
    const { container } = render(<MenuContact />)

    expect(container).toMatchSnapshot()
  })
})
