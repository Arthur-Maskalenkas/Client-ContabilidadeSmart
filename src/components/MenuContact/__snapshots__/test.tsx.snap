// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuContact /> should render the heading 1`] = `
.c0 {
  border-bottom: 1px solid #3F4350;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2rem;
  padding-left: 2rem;
  height: 100%;
}

.c3:nth-child(2) {
  border-right: 1px solid #3F4350;
  border-left: 1px solid #3F4350;
}

.c3 .c4,
.c3 .c7 {
  cursor: pointer;
  -webkit-transition: 0.5s all;
  transition: 0.5s all;
}

.c3 .c4:hover {
  background-color: #F66A3E;
}

.c3 .c7:hover {
  color: #F66A3E;
}

.c8 {
  color: #FFFFFF;
  font-size: 2rem;
  font-weight: 300;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.2rem;
  border-radius: 50%;
  margin-right: 0.8rem;
  background: rgb(255 255 255 / 9%);
}

.c5 > svg {
  width: 2rem;
  color: white;
}

.c2 {
  position: relative;
  height: 7.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem);
  padding-right: calc(3.2rem);
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4 c5"
          >
            <svg
              class="c6"
              fill="currentColor"
              focusable="false"
              role="svg"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                whatsapp
              </title>
              <path
                d="M13.601 2.326A7.854 7.854 0 0 0 7.994 0C3.627 0 .068 3.558.064 7.926c0 1.399.366 2.76 1.057 3.965L0 16l4.204-1.102a7.933 7.933 0 0 0 3.79.965h.004c4.368 0 7.926-3.558 7.93-7.93A7.898 7.898 0 0 0 13.6 2.326zM7.994 14.521a6.573 6.573 0 0 1-3.356-.92l-.24-.144-2.494.654.666-2.433-.156-.251a6.56 6.56 0 0 1-1.007-3.505c0-3.626 2.957-6.584 6.591-6.584a6.56 6.56 0 0 1 4.66 1.931 6.557 6.557 0 0 1 1.928 4.66c-.004 3.639-2.961 6.592-6.592 6.592zm3.615-4.934c-.197-.099-1.17-.578-1.353-.646-.182-.065-.315-.099-.445.099-.133.197-.513.646-.627.775-.114.133-.232.148-.43.05-.197-.1-.836-.308-1.592-.985-.59-.525-.985-1.175-1.103-1.372-.114-.198-.011-.304.088-.403.087-.088.197-.232.296-.346.1-.114.133-.198.198-.33.065-.134.034-.248-.015-.347-.05-.099-.445-1.076-.612-1.47-.16-.389-.323-.335-.445-.34-.114-.007-.247-.007-.38-.007a.729.729 0 0 0-.529.247c-.182.198-.691.677-.691 1.654 0 .977.71 1.916.81 2.049.098.133 1.394 2.132 3.383 2.992.47.205.84.326 1.129.418.475.152.904.129 1.246.08.38-.058 1.171-.48 1.338-.943.164-.464.164-.86.114-.943-.049-.084-.182-.133-.38-.232z"
              />
            </svg>
          </div>
          <a
            class="c7 c8"
          >
            +55 17 99146-0296
          </a>
        </div>
        <div
          class="c3"
        >
          <div
            class="c4 c5"
          >
            <svg
              class="c6"
              fill="currentColor"
              focusable="false"
              role="svg"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Gmail
              </title>
              <path
                d="M24 5.457v13.909c0 .904-.732 1.636-1.636 1.636h-3.819V11.73L12 16.64l-6.545-4.91v9.273H1.636A1.636 1.636 0 0 1 0 19.366V5.457c0-2.023 2.309-3.178 3.927-1.964L5.455 4.64 12 9.548l6.545-4.91 1.528-1.145C21.69 2.28 24 3.434 24 5.457z"
              />
            </svg>
          </div>
          <a
            class="c7 c8"
          >
            contato@contabilidadesmart.com.br
          </a>
        </div>
        <div
          class="c3"
        >
          <div
            class="c4 c5"
          >
            <svg
              class="c6"
              fill="currentColor"
              focusable="false"
              role="svg"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Instagram
              </title>
              <path
                d="M8 0C5.829 0 5.556.01 4.703.048 3.85.088 3.269.222 2.76.42a3.917 3.917 0 0 0-1.417.923A3.927 3.927 0 0 0 .42 2.76C.222 3.268.087 3.85.048 4.7.01 5.555 0 5.827 0 8.001c0 2.172.01 2.444.048 3.297.04.852.174 1.433.372 1.942.205.526.478.972.923 1.417.444.445.89.719 1.416.923.51.198 1.09.333 1.942.372C5.555 15.99 5.827 16 8 16s2.444-.01 3.298-.048c.851-.04 1.434-.174 1.943-.372a3.916 3.916 0 0 0 1.416-.923c.445-.445.718-.891.923-1.417.197-.509.332-1.09.372-1.942C15.99 10.445 16 10.173 16 8s-.01-2.445-.048-3.299c-.04-.851-.175-1.433-.372-1.941a3.926 3.926 0 0 0-.923-1.417A3.911 3.911 0 0 0 13.24.42c-.51-.198-1.092-.333-1.943-.372C10.443.01 10.172 0 7.998 0h.003zm-.717 1.442h.718c2.136 0 2.389.007 3.232.046.78.035 1.204.166 1.486.275.373.145.64.319.92.599.28.28.453.546.598.92.11.281.24.705.275 1.485.039.843.047 1.096.047 3.231s-.008 2.389-.047 3.232c-.035.78-.166 1.203-.275 1.485a2.47 2.47 0 0 1-.599.919c-.28.28-.546.453-.92.598-.28.11-.704.24-1.485.276-.843.038-1.096.047-3.232.047s-2.39-.009-3.233-.047c-.78-.036-1.203-.166-1.485-.276a2.478 2.478 0 0 1-.92-.598 2.48 2.48 0 0 1-.6-.92c-.109-.281-.24-.705-.275-1.485-.038-.843-.046-1.096-.046-3.233 0-2.136.008-2.388.046-3.231.036-.78.166-1.204.276-1.486.145-.373.319-.64.599-.92.28-.28.546-.453.92-.598.282-.11.705-.24 1.485-.276.738-.034 1.024-.044 2.515-.045v.002zm4.988 1.328a.96.96 0 1 0 0 1.92.96.96 0 0 0 0-1.92zm-4.27 1.122a4.109 4.109 0 1 0 0 8.217 4.109 4.109 0 0 0 0-8.217zm0 1.441a2.667 2.667 0 1 1 0 5.334 2.667 2.667 0 0 1 0-5.334z"
              />
            </svg>
          </div>
          <a
            class="c7 c8"
          >
            @contabilidadesmart
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
